### PR TITLE
[TF-750] Move prop fetching logic for pages into the Theme Kit

### PR DESCRIPTION
**Pull request by oleg-semyonov**
TODO: Add documentation on the helper methods.

The code is basically taken from the Bea theme, with some wrapping logic to allow for passing custom options.

### CODE REVIEW

File: src/adapter-nextjs/page-props/category.ts
Comment (oleg-semyonov) [inline src/adapter-nextjs/page-props/category.ts:69]: In some cases the custom page props depend on the request context or `newsroomContextProps`, this allows to pass a function that derives custom props from these.

**Review by e1himself — COMMENTED** [review]
I've reviewed this PR. The code looks good.

However, I don't have much of an idea of what's going on here. Totally unfamiliar with the rest of the codebase.

So it would be great to get a review from one of the "maintainers" :)

File: src/adapter-nextjs/page-props/category.ts
Comment (kudlajz) [inline src/adapter-nextjs/page-props/category.ts:17]: Why not just call this the same as the interface - `extraStoryFields`? Seems more readable.

**Review by kudlajz — APPROVED** [review]

File: src/adapter-nextjs/page-props/category.ts
Comment (oleg-semyonov) [reply src/adapter-nextjs/page-props/category.ts:17]: Yeah, that makes much more sense! Thank you!

**Pull request merged**
